src/main/java/org/folio/innreach/domain/service/impl/RecordContributionServiceImpl.java

```diff
@@ -121,9 +132,19 @@ private InnReachResponse verifyBibContribution(UUID centralServerId, String bibI
   }
 
   private InnReachResponse contributeBibItems(String bibId, UUID centralServerId, List<BibItem> bibItems) {
-    var response = irContributionService.contributeBibItems(centralServerId, bibId, BibItemsInfo.of(bibItems));
-    Assert.isTrue(response.isOk(), "Unexpected items contribution response: " + response);
-    return response;
+    try {
+      var response = irContributionService.contributeBibItems(centralServerId, bibId, BibItemsInfo.of(bibItems));
+      if (!response.getErrors().isEmpty()) {
+        var error = response.getErrors().get(0).getReason();
+        if (error.contains("Contribution to d2irm is currently suspended")) {
+          throw new ServiceSuspendedException("Contribution to d2irm is currently suspended");
+        }
+      }
```

Comment:
This block is duplicated with the block above in `contributeBib`. Code duplication is always bad. Abstract this block into a method on the response object called `checkServiceSuspension` which throws our exception.